Modules/_ssl.c

```diff
@@ -5514,7 +5534,8 @@ PySSLSession_richcompare(PyObject *left, PyObject *right, int op)
     }
 
     int result;
-    PyTypeObject *sesstype = ((PySSLSession*)left)->ctx->state->PySSLSession_Type;
+    _sslmodulestate *state = get_state_obj(left);
+    PyTypeObject *sesstype = state->PySSLSession_Type;
```

Comment:
Is this better than `PySSLSession_CAST(left)->ctx->state->PySSLSession_Type`?

Reply:
At this point, we don't know whether `left` is of correct type strictly speaking, so the cast may be incorrect.

Reply:
If `left` is not the correct type, `get_state_obj` will misbehave too. It relies on the object being an exact type (not subtype) defined in the ssl module.

IMO, it's better to use the macro that might get an assert in the future.
```suggestion
    PyTypeObject *sesstype = PySSLSession_CAST(left)->ctx->state->PySSLSession_Type;
```

Reply:
Oh, I thought it would have worked a bit more magically! I guess we can indeed do the cast. Feel free to edit the PR as I won't be on my PC for 10 days!

Reply:
OK.
Looking a bit closer: Since session is not subclassable, and `tp_richcompare` guarantees the first argument has the right type, we can just use `Py_TYPE(left)`.